firebase-firestore/src/test/java/com/google/firebase/firestore/FilterTest.java

```diff
@@ -0,0 +1,188 @@
+    assertNotEquals(filter, Filter.arrayContainsAny("x", ImmutableList.of("v2", "v3")));
+    assertNotEquals(filter, Filter.arrayContainsAny("y", ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.equalTo("y", "v"));
+  }
+
+  @Test
+  public void inArray() {
+    Filter filter = Filter.inArray("x", ImmutableList.of("v1", "v2"));
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.inArray(FieldPath.of("x"), ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.inArray("x", ImmutableList.of("v2", "v3")));
+    assertNotEquals(filter, Filter.inArray("y", ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.notInArray("y", ImmutableList.of("v1", "v2")));
+  }
+
+  @Test
+  public void notInArray() {
+    Filter filter = Filter.notInArray("x", ImmutableList.of("v1", "v2"));
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.notInArray(FieldPath.of("x"), ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.notInArray("x", ImmutableList.of("v2", "v3")));
+    assertNotEquals(filter, Filter.notInArray("y", ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.inArray("y", ImmutableList.of("v1", "v2")));
+  }
+
+  @Test
+  public void or() {
+    Filter filter = Filter.or(
+        Filter.equalTo("x", ImmutableList.of("v1", "v2")),
+        Filter.equalTo("y", ImmutableList.of("v3", "v4"))
+    );
+    assertEquals(
+        filter,
+        Filter.or(
+            Filter.equalTo(FieldPath.of("x"), ImmutableList.of("v1", "v2")),
+            Filter.equalTo(FieldPath.of("y"), ImmutableList.of("v3", "v4"))
+        )
+    );
+    assertNotEquals(
+        filter,
+        Filter.and(
+            Filter.equalTo("x", ImmutableList.of("v1", "v2")),
+            Filter.equalTo("y", ImmutableList.of("v3", "v4"))
+        )
+    );
+    assertNotEquals(
+        filter,
+        Filter.and(
+            Filter.equalTo("y", ImmutableList.of("v3", "v4")),
+            Filter.equalTo("x", ImmutableList.of("v1", "v2"))
+        )
+    );
+  }
+
+  @Test
+  public void and() {
+    Filter filter = Filter.and(
+        Filter.equalTo("x", ImmutableList.of("v1", "v2")),
+        Filter.equalTo("y", ImmutableList.of("v3", "v4"))
+    );
+    assertEquals(
+        filter,
+        Filter.and(
+            Filter.equalTo(FieldPath.of("x"), ImmutableList.of("v1", "v2")),
+            Filter.equalTo(FieldPath.of("y"), ImmutableList.of("v3", "v4"))
+        )
+    );
+    assertNotEquals(
+        filter,
+        Filter.or(
+            Filter.equalTo("x", ImmutableList.of("v1", "v2")),
+            Filter.equalTo("y", ImmutableList.of("v3", "v4"))
+        )
+    );
+    assertNotEquals(
+        filter,
+        Filter.or(
+            Filter.equalTo("y", ImmutableList.of("v3", "v4")),
+            Filter.equalTo("x", ImmutableList.of("v1", "v2"))
+        )
+    );
```

Comment:
add the same two cases as `or()` test above.

Reply:
Already there. See last assert.

Reply:
Ignore above comment. Will do.